### ISSUE TITLE
Missing `IO::READABLE`, `IO::WRITABLE` and `IO::PRIORITY`.

### ISSUE THREAD

**Issue by ioquatix** [jruby/jruby#7512]
```
> ruby --version
jruby 9.4.0.0 (3.1.0) 2022-11-23 95c0ec159f OpenJDK 64-Bit Server VM 19.0.1+10 on 19.0.1+10 +jit [x86_64-linux]
```

Missing `IO::READABLE`, `IO::WRITABLE` and `IO::PRIORITY`.

Thanks

**Comment by headius** [comment]
Safe to say this is specific to the new scheduler API? We will do that work in a batch, so I want to keep the issues associated.

**Comment by ioquatix** [comment]
It's used for `IO#wait` which can take an integer mask. It was probably introduced for the fiber scheduler IIRC.

**Comment by headius** [comment]
I can add these but io/wait will take some time to update since we did not ship with these constants/enums defined in 9.4.

Did we ever agree to merge io/wait into core or was that io/nonblock?

**Comment by ioquatix** [comment]
`io/wait` was merged, and `io/nonblock` was left as an extension.

I tried to work on merging `io/nonblock` but after implementing it, I found it was tricky to get parity with Windows.

Because of this, I felt less confident about merging `io/nonblock` into core.

Looking at the big picture, things like `io.nonblock` and related interfaces are mostly there as an artefact of people trying to implement their own non-blocking IO. The reality is, this kind of design is not composable. So, ultimately, I felt like `io/nonblock` is a poor interface, because it encourages bad code, basically.

For me, it made me think about how to achieve those operations in Async without any such interface, and I think for the most part, I fall back to the mental model of "The user shouldn't have to care". IOW, whether an IO is non-blocking or not shouldn't impact the concurrency/scheduling of a blocking `io.read` or `io.write` operation.

I'm happy to elaborate on it more if this is something you want to discuss.
